### ISSUE TITLE
Style/GuardClause false negative

### ISSUE THREAD

**Issue by GeorgeMarcusGG** [rubocop/rubocop#10904]
## Code

```ruby
def foo
  if true
    1
  else
    2
  end
end
```

## Expected behavior

The cop should have been triggered to suggest this change
```ruby
def foo
  return 1 if true

  2
end
```

This is especially important if the `else` branch has lots of lines.

## Actual behavior

No trigger.

If I remove the `else` branch then the cop is triggered.
```ruby
def foo
  if true
    1
  end
end
```

If I add a `return` to each branch then the cop is triggered.
```ruby
def foo
  if true
    return 1
  else
    return 2
  end
end
```

## Steps to reproduce the problem

Code above

## RuboCop version

```
bundle exec rubocop -V
1.34.1 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.1 x86_64-darwin21)
```

**Comment by ydah** [comment]
That behavior seems correct. `Style/GuardClause` cop is offense for conditional branches with if-else if the body of the condition contains `return` or `break` or `next` or `raise` or `fail`.

Therefore, the following code is not an offense:
```ruby
def func
  if something
    work
  else
    test
  end
end

def func
  unless something
    work
  else
    test
  end
end
```

However, the following code is an offense:
```ruby
def func
  if something
    work
  else
    test || raise('message')
  end
end

def func
  if something
    work and return
  else
    test
  end
end
```

**Comment by GeorgeMarcus** [comment]
Ok. I was just thinking that in this case the `if` is at the end of the function so both branches have an implicit `return`.

Also I don't understand why having or not having an `else` branch should make any difference.

If this is not intended or if you don't think this would be useful then feel free to close the issue.

**Comment by bbatsov** [comment]
That's by design indeed - sometimes an `if`/`else` just reads better, so we wanted to leave people the option to chose what go for. I recall originally this cop was always enforcing the style you prefer, perhaps we left this as an option (or can add an option). At any rate - I want this cop to be relatively relaxed out-of-the-box.

**Comment by bbatsov** [comment]
Btw, I see that the cop examples are clearly incorrect - e.g. we have this very same code listed as something that would be marked as an offense and most of the listings are not wrapped in the `def`s. We'll need to fix this.
